### PR TITLE
Add onclick to action element

### PR DESCRIPTION
**Pull request by am26090**
New PR - https://github.com/mi6/ic-ui-kit/pull/2986/commits

### CODE REVIEW

**Comment by MI6-255** [comment]
Thanks for your contribution, you just need to run prettier to fix the broken GitHub checks.

**Comment by am26090** [comment]
> Thanks for your contribution, you just need to run prettier to fix the broken GitHub checks, then a member of the team will take a look.

Should be done

**Review by ad9242 — CHANGES_REQUESTED** [review]
It would be good to add some documentation about the actionElement and actionOnClick. This would make it much clearer for users as at the moment you have to look into the code or examples to discover that this functionality exists.

Maybe as a new story, after the "Links and Elements in data" story (ideally in both canary-web-components & canary-react) along with a paragraph describing the functionality (and maybe why users might want to go with this approach rather than using slotted elements)

File: packages/canary-react/src/stories/ic-data-table.stories.mdx [hunk 1]
```diff
@@ -2114,6 +2114,7 @@ const DATA = [
     firstName: {
       data: "Joe",
       actionElement: `<ic-button variant="icon" size="small  aria-label="you can disable tooltips on icon buttons">  <svg  aria-label="refresh button"  xmlns="http://www.w3.org/2000/svg"    width="24"    height="24"    viewBox="0 0 24 24"    fill="#000000"  >    <path d="M0 0h24v24H0V0z" fill="none"></path>    <path d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"></path>  </svg></ic-button>`,
+      actionOnClick: () => console.log('hello)
```
Comment (ad9242) [inline packages/canary-react/src/stories/ic-data-table.stories.mdx:2117]:
```suggestion
      actionOnClick: () => console.log('hello')
```

**Review by ad9242 — CHANGES_REQUESTED** [review]
Please can you also add the sample code to the canary-web-components Action Element story (like other stories above it in the file)?

**Review by ad9242 — CHANGES_REQUESTED** [review]
Please can you run `npm run prettier:fix` and commit the changes?

Also please check that the Cypress tests are running\passing locally, as they seem to be failing in the CI (none of the data-table tests are running, due to an error in the file it seems.

Please can you also squash the commits? There should be only 1 commit at each scope: "feat(canary-web-components)........" (note "s" on end) and "feat(canary-react) .......". So this Pull Request would then only contain 2 commits.

**Comment by am26090** [comment]
> Please can you run `npm run prettier:fix` and commit the changes?
> 
> Also please check that the Cypress tests are running\passing locally, as they seem to be failing in the CI (none of the data-table tests are running, due to an error in the file it seems.
> 
> Please can you also squash the commits? There should be only 1 commit at each scope: "feat(canary-web-components)........" (note "s" on end) and "feat(canary-react) .......". So this Pull Request would then only contain 2 commits.

ive tried to get it down to 2 commits, goes wrong every time. So ive got it to 3, if thats ok

**Comment by ad9242** [comment]
I think we will need to get that commit removed I'm afraid. One option would be to create a new branch off the latest develop & then cherry-pick your other commits:

eg
```
git cherry-pick bfa8d62942887272d5c9f0be51d3f7752c94b907
git cherry-pick 702d9fc62d0578778626287e463ed3518d82469b
```
and then create a new PR for that branch

it looks like this also needs a `npm run:prettier:fix` too to resolve some formatting issues

**Comment by am26090** [comment]
> I think we will need to get that commit removed I'm afraid. One option would be to create a new branch off the latest develop & then cherry-pick your other commits:
> 
> eg
> 
> ```
> git cherry-pick bfa8d62942887272d5c9f0be51d3f7752c94b907
> git cherry-pick 702d9fc62d0578778626287e463ed3518d82469b
> ```
> 
> and then create a new PR for that branch
> 
> it looks like this also needs a `npm run:prettier:fix` too to resolve some formatting issues

new pr, see here https://github.com/mi6/ic-ui-kit/pull/2986/commits
